operator/pkg/ingress/ingress_reconcile.go

```diff
@@ -276,6 +281,13 @@ func (r *ingressReconciler) buildDedicatedResources(ctx context.Context, ingress
 		}
 	}
 
+	eTP, err := annotations.GetAnnotationServiceExternalTrafficPolicy(ingress)
+	if err != nil {
+		scopedLog.WithError(err).Error("failed to get externalTrafficPolicy annotation from Ingress object")
```

Comment:
If we only want to warn the user about an invalid annotation value (but continue with the logic) - i suggest to use `Warn` instead of `Error`. 

Referring to one of your questions from the PR description

> Setting an invalid ingress.cilium.io/service-external-traffic-policy value on an ingress.cilium.io/loadbalancer-mode=dedicated Ingress resource "spams" the Cilium Operator logs with multiple (~10) Reconciler error messages. Even after the initial apply, we'll see such an error every other few seconds/minutes. Is there any need to mitigate this? I don't think so, but just checking.

IMO then it's also OK to "spam" the operator log with these warnings.

Note: Log messages should start with Uppercase whereas it's a Go convention to keep errors lowercase -> Change to start uppercase